ReactWindows/ReactNative.Shared/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,229 @@
+            if (IsHiddenElement(uiElementAutomationPeer))
+            {
+                return;
+            }
+
+            // Update AccessibilityView property accordingly.
+            switch (importantForAccessibilityNew)
+            {
+                case ImportantForAccessibility.Auto:
+                    uiElement.ClearValue(AutomationProperties.AccessibilityViewProperty);
+                    SetChildrenAccessibilityViewFromImportantForAccessibility(uiElementAutomationPeer);
+                    break;
+                case ImportantForAccessibility.No:
+                    AutomationProperties.SetAccessibilityView(uiElement, AccessibilityView.Raw);
+                    SetChildrenAccessibilityViewFromImportantForAccessibility(uiElementAutomationPeer);
+                    break;
+                case ImportantForAccessibility.NoHideDescendants:
+                    AutomationProperties.SetAccessibilityView(uiElement, AccessibilityView.Raw);
+                    SetChildrenAccessibilityView(uiElementAutomationPeer, AccessibilityView.Raw);
+                    break;
+                case ImportantForAccessibility.Yes:
+                    AutomationProperties.SetAccessibilityView(uiElement, AccessibilityView.Content);
+                    SetChildrenAccessibilityView(uiElementAutomationPeer, AccessibilityView.Raw);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Attached property used to store ImportantForAccessibility value in native controls.
+        /// </summary>
+        public static readonly DependencyProperty ImportantForAccessibilityAttachedProperty = DependencyProperty.RegisterAttached(
+                                                                                                "ImportantForAccessibilityAttached",
+                                                                                                typeof(ImportantForAccessibility),
+                                                                                                typeof(UIElement),
+                                                                                                new PropertyMetadata(ImportantForAccessibility.Auto));
+
+        /// <summary>
+        /// ImportantForAccessibilityAttached property setter.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <param name="value"></param>
+        public static void SetImportantForAccessibilityAttached(UIElement element, ImportantForAccessibility value)
+        {
+            element.SetValue(ImportantForAccessibilityAttachedProperty, value);
+        }
+
+        /// <summary>
+        /// ImportantForAccessibilityAttached property getter.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <returns></returns>
+        public static ImportantForAccessibility GetImportantForAccessibilityAttached(UIElement element)
```

Comment:
>UIElement [](start = 85, length = 9)

GetValue is a DependencyObject's method. why do we restrict it to be UIElement?

Reply:
True. But elements if the implementation assume UIElement, e.g. FrameworkElementAutomationPeer.Owner, so I thought UIElement is high enough and common enough to base implementation on it, instead of on DependencyObject 

---
In reply to: [163100620](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163100620) [](ancestors = 163100620)